### PR TITLE
Jetpack Backup: Update component to use custom backup retention days if set

### PR DESCRIPTION
**Pull request by rustamshaikh**
#### Proposed Changes

* Update the component to make use of a custom backup retention period (in days) if set by the customer otherwise use the retention from the plan.

Requires D100421-code on VP.

#### Testing Instructions

1) Take a site whose backups are being discarded due to the storage limit has been reached.
2) Check the current backup dashboard status, it should be similar to the below screenshot. Ensure that no custom retention has been set yet or delete if it is, from the vpshell.
     Screenshot 2023-02-07 at 5 49 05 PM
3) Now set a custom retention period lower than that of the current `days_of_backups_allowed` from the size API either by following D100335-code or via from vpshell (e.g. in D100421-code)
4) Refresh the Backup Dashboard on Calypso, it should show normal usage with no warning message as below:
     Screenshot 2023-02-07 at 5 55 28 PM

Also, ensure unit tests passes by executing `yarn run test-client client/state/rewind`

#### Pre-merge Checklist

- [x] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [x] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [x] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [x] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

Related to D100421-code

### CODE REVIEW

File: client/state/rewind/test/selectors.js [hunk 1]
```diff
@@ -33,3 +33,38 @@ describe( 'getRewindStorageUsageLevel()', () => {
...
+							retentionDays: 30,
+						},
+					},
+				},
+			},
+			siteId: 123,
+			expected: 30,
+		},
+	] )(
+		'should return the retentionDays if passed, null otherwise',
```
Comment (jwebbdev) [inline client/state/rewind/test/selectors.js]:
```suggestion
		'should return the retentionDays if passed, undefined otherwise',
```

**Review by jwebbdev — COMMENTED** [review]
Tested out on one of my test sites and worked.  Minor comment on a test description.

**Review by Initsogar — COMMENTED** [review]
It looks good to me. You will just need to resolve conflicts introduced due to a recent PR (#73031) that we recently merged that touched the same files.

> Refresh the Backup Dashboard on Calypso, it should show normal usage with no warning message as below:

Also, just wondering if we should update the `X days of saved backups` text there with the lower retention period we are setting. Any thoughts?

**Comment by rustamshaikh** [comment]
> It looks good to me. You will just need to resolve conflicts introduced due to a recent PR (#73031) that we recently merged that touched the same files.
> 
> > Refresh the Backup Dashboard on Calypso, it should show normal usage with no warning message as below:
> 
> Also, just wondering if we should update the `X days of saved backups` text there with the lower retention period we are setting. Any thoughts?

The `X days of saved backups` text will get updated once the next daily backup is run, are you suggesting that we can show the updated message even though it has not yet been discarded (updated the saved backup count) in the backend 🤔 . Maybe we can if the new retention is lower than the original days of backups saved, we can show this safely like `(new retention days number) days of saved backups` since anyway, it will reduce the no. of saved backups to the new lower retention after next daily backup.

**Comment by Initsogar** [comment]
> The `X days of saved backups` text will get updated once the next daily backup is run

Thanks for clarification! Indeed, that's the expected behavior.

> are you suggesting that we can show the updated message even though it has not yet been discarded (updated the saved backup count) in the backend 🤔 . Maybe we can if the new retention is lower than the original days of backups saved, we can show this safely like `(new retention days number) days of saved backups` since anyway, it will reduce the no. of saved backups to the new lower retention after next daily backup.

I was just overthinking about how to present the current selection to the user, but I think we can still manage that on the setting. I'm taking notes and I will raise the question to Design.

**Review by Initsogar — APPROVED** [review]
Looks good to me! 👍🏽

**Pull request merged**